core/src/Borrower.sol

```diff
@@ -111,8 +111,7 @@ contract Borrower is IUniswapV3MintCallback {
         TOKEN0 = lender0.asset();
         TOKEN1 = lender1.asset();
 
-        require(pool.token0() == address(TOKEN0));
-        require(pool.token1() == address(TOKEN1));
+        assert(pool.token0() == address(TOKEN0) && pool.token1() == address(TOKEN1));
```

Comment:
What is the difference between assert and require?

Reply:
require is usually used for checking external inputs, assert is for internal stuff. Since this data is coming from the Factory, I'm considering it internal.

I debated taking out the check entirely too.